tests/phpunit/tests/blocks/register.php

```diff
@@ -431,6 +431,71 @@ public function test_metadata_not_found_in_the_current_directory() {
 		$this->assertFalse( $result );
 	}
 
+	/**
+	 * Test registering a block using arguments instead of a block.json file.
+	 *
+	 * @ticket 56865
+	 */
+	public function test_register_block_type_from_metadata_with_arguments() {
+		$result = register_block_type_from_metadata(
+			'',
+			array(
+				'api_version' => 2,
+				'name'        => 'tests/notice-from-array',
+				'title'       => 'Notice from array',
+				'category'    => 'common',
+				'icon'        => 'star',
+				'description' => 'Shows warning, error or success notices… (registered from an array)',
+				'keywords'    => array(
+					'alert',
+					'message',
+				),
+				'textdomain'  => 'notice-from-array',
+			)
+		);
+
+		$this->assertInstanceOf( 'WP_Block_Type', $result );
```

Comment:
Just need a final `$message` parameter to explain what went wrong when each of these assertions fails - same for the other new test method.

Reply:
Hmmm what kind of message could we add here? 
If the test fails, it will already print `Failed asserting that null is an instance of class "WP_Block_Type"`, which is pretty self-explanatory IMO
That's just what all pre-existing tests do in this file, should we also add messages to all other pre-existing checks in this file? 🤔

Reply:
Admittedly, in general this type of assertion would be self-explanatory (assuming that a given test provides enough context about why the assertion is needed). Test methods with multiple assertion's need a `$message` parameter for each assertion per the standards:

> All PHPUnit assertions, as well as all WordPress custom assertions, allow for a $message parameter to be passed. This message will be displayed when the assertion fails and can help immensely when debugging a test. This parameter should always be used if more than one assertion is used in a test method.
> [Ref](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions)

For this test as written, I'd suggest something like:
- "The block was not registered."
- "The API version is incorrect."
- "The block's name is incorrect."
- etc.

Reply:
Regarding the other tests in this file that weren't introduced in this PR, we can either decide to add the `$message` for this PR's tests to meet standards, and follow-up on the rest at a later time, or do all of the tests at once at a later time. This comes down to a preference of whether to reduce future work now, or do it all at once. I don't have a strong opinion on what's preferred, so I'm happy to go with your preference on this one.

Reply:
Fair point, and thank you for the explanation!
I added messages to the new tests here.
We can address the older ones in a follow-up PR since they are not directly related to this PR 👍